edge/pkg/edgestream/edgestream.go

```diff
@@ -73,27 +73,32 @@ func (e *edgestream) Start() {
 		Path:   "/v1/kubeedge/connect",
 	}
 	// TODO: Will improve in the future
-	ok := <-edgehub.HasTLSTunnelCerts
-	if ok {
-		cert, err := tls.LoadX509KeyPair(config.Config.TLSTunnelCertFile, config.Config.TLSTunnelPrivateKeyFile)
-		if err != nil {
-			klog.Exitf("Failed to load x509 key pair: %v", err)
-		}
-		tlsConfig := &tls.Config{
-			InsecureSkipVerify: true,
-			Certificates:       []tls.Certificate{cert},
-		}
+	if ok := <-edgehub.HasTLSTunnelCerts; !ok {
+		klog.Exitf("Failed to write cert key pair into disk")
+	}
```

Comment:
What needs to be achieved here is the processing without the certificate, can you try it?

Reply:
The happy path is left-aligned, so log and exit when ok is not true.

Reply:
What should we do if no certs? http instead of https?

Reply:
Should exits the process, I think you are right.

Reply:
```golang
func (eh *EdgeHub) Start() {
    ...
    HasTLSTunnelCerts <- true
    close(HasTLSTunnelCerts)
```
In the edgehub module, it is hard-coded, and this channel will only receive `true` forever.

Reply:
Like `TLSClientConnect` method.
You can try to implement a `ws` connection instead of a secure `wss` connection.
However, we need to consider whether non secure connections are necessary?
@hainesc @fisherxu

Reply:
It's the cloud-edge connection, I think secure wss is needed. :)

Reply:
@zc2638 The code here should not dependent the hard code of the channel `HasTLSTunnelCerts`, as @fisherxu says, if we should exits the process if we get `false` from the channel, we should exit here. The hard code is a temporary work and somebody may improve it.

Reply:
Should we remove the `HasTLSTunnelCerts ` channel? 
Since it must be `true`, I think there is no need to exist here.

Reply:
Through the exit will not be triggered, but I think it's more reasonable to add  the exit codes. If there are some bugs due to code modification, it can exit here.

But the exit message seems wrong. Should be `Failed to obtain certs`?

Reply:
Since it has been determined that only TLS is supported, we should delete the channel logic for judging TLS.
The following logic is sufficient to determine whether the certificate exists and is valid.
```golang
cert, err := tls.LoadX509KeyPair(config.Config.TLSTunnelCertFile, config.Config.TLSTunnelPrivateKeyFile)
if err != nil {
    klog.Exitf("Failed to load x509 key pair: %v", err)
}
```

Reply:
Since it has been determined that only TLS is supported, we should delete the channel logic for judging TLS.
The following logic is sufficient to determine whether the certificate exists and is valid.
```golang
cert, err := tls.LoadX509KeyPair(config.Config.TLSTunnelCertFile, config.Config.TLSTunnelPrivateKeyFile)
if err != nil {
    klog.Exitf("Failed to load x509 key pair: %v", err)
}
```

Reply:
I agree with @fisherxu

Reply:
@fisherxu exit message modified.

Reply:
> Since it has been determined that only TLS is supported, we should delete the channel logic for judging TLS.
The following logic is sufficient to determine whether the certificate exists and is valid.

No, the certs is generated in edgehub module, so we must wait edgehub completed, then trigger here to load.